Recipes/Controller/MainViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  MainViewController.swift
+//  Recipes
+//
+//  Created by krikaz on 3/12/20.
+//  Copyright © 2020 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class MainViewController: UIViewController {
+    
+    let networkClient = RecipesNetworkClient()
+    
+    var allRecipes: [Recipe] = [] {
+        didSet {
+            filterRecipes()
+        }
+    }
+    
+    var recipesTableViewController: RecipesTableViewController? {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    var filteredRecipes: [Recipe] = [] {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    @IBOutlet weak var searchField: UITextField!
+    @IBAction func searchAction(_ sender: UITextField) {
+        searchField.resignFirstResponder()
+        filterRecipes()
+    }
+    
+    func filterRecipes() {
+        if let searchWords = searchField.text {
+            filteredRecipes = allRecipes.filter { $0.name.contains(searchWords) || $0.instructions.contains(searchWords) }
+        } else {
+            filteredRecipes = allRecipes
+        }
+    }
+    
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        networkClient.fetchRecipes { recipes, error in
+            if let error = error {
+                NSLog("Error fetching recipes: \(error)")
+                return
+            }
+            
+            DispatchQueue.main.async {
+                if let recipes = recipes {
+                    self.allRecipes = recipes
+                }
+            }
+        }
```

Comment:
In real production app, this kind of code would not be in the ViewController since its responsibility is to handle the view.